tools/schemacode/bidsschematools/validator.py

```diff
@@ -600,55 +600,57 @@ def select_schema_dir(
     for bids_path in bids_paths:
         bids_path = os.path.abspath(os.path.expanduser(bids_path))
         dataset_description = _find_dataset_description(bids_path)
-        if dataset_description:
-            if dataset_description in dataset_descriptions:
-                raise ValueError(
-                    "You have selected files belonging to 2 different datasets. "
-                    "Please run the validator once per dataset."
+        if dataset_description not in dataset_descriptions:
+            dataset_descriptions.append(dataset_description)
+    if len(dataset_descriptions) > 1:
+        raise ValueError(
+            f"You have selected files belonging to {len(dataset_descriptions)} "
+            "different datasets. Please run the validator once per dataset."
+        )
```

Comment:
what if there is a BIDS subdaset under `rawdata/` or `derivatives/` -- wouldn't we find multiple dataset_descriptions?

Reply:
whelp, your're right... we do filter those out but we filter them *after* we detect the schema in the path getting function... might end up having to merge this PR with https://github.com/bids-standard/bids-specification/issues/1155 --- which I've also been meaning to tackle for a while.

Reply:
@yarikoptic ok, so this is not actually happening, as the dataset_descriptor finding simply crawls up until if finds a file. So https://github.com/bids-standard/bids-specification/blob/73db3bafe3317645d1b72d1798308f85a5efd238/tools/schemacode/bidsschematools/validator.py#L602 will only ever find one or zero files per iteration.

Reply:
so it would error out only if I point with the `paths` to multiple BIDS datasets, e.g. `[".", "rawdata"]` right?

then may be adjust docstring for `bids_paths` to say that they all must be within the same BIDS dataset, i.e. should not e.g. point to main BIDS dataset and derivatives etc.

Reply:
@yarikoptic done.